### PR TITLE
[ceca] Report number of correctly guessed key byte differences

### PR DESCRIPTION
**Pull request by vogelpi**
This number is a actually a far more relevant indicator for the amount of leakage in the traces than successful key recovery (pass/fail). Therefore, this PR adds the functionality to compute this number and report it on the command line.

This solves #30 .

### CODE REVIEW

File: cw/cw305/ceca.py [hunk 1]
```diff
@@ -408,6 +407,34 @@ def recover_key(diffs, attack_direction, plaintext, ciphertext):
...
+    """Compares pairwise_diffs_scores with diffs between bytes in correct_key.
+
+    This function takes the differences between key bytes computed in
+    pairwise_diffs_scores and compares them with the actual differences in the
+    correct key.
+
+    Args:
+        pairwise_diffs_scores: A 16x16x2 matrix of pairwise differences between
+            key bytes and their confidence scores.
+        attack_direction: Attack direction, 'input' or 'output'.
```
Comment (alphan) [inline cw/cw305/ceca.py:420]: nit: "'input' or 'output'" reads like `attack_direction` is a string.

**Review by alphan — APPROVED** [review]
Thanks @vogelpi!

File: cw/cw305/ceca.py (hunk 1 above)
Comment (vogelpi) [reply cw/cw305/ceca.py:420]: Hey Alphan, thanks for your review!

There are too many quotes here, sorry. How exactly would you like me to change this?

Comment (alphan) [reply cw/cw305/ceca.py:420]: Ah, sorry, I was thinking maybe we should remove that part:

```suggestion
        attack_direction: Attack direction.
```

Comment (vogelpi) [reply cw/cw305/ceca.py:420]: Oh, I see. I can do that. Thanks for the clarification.

**Pull request merged**